src/GreenPipes/IContextSource.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace GreenPipes
+{
+    public interface IContextSource<TContext> where TContext : class, PipeContext
+    {
+        Task Fill(IPipe<TContext> pipe, CancellationToken token = default(CancellationToken));

```

Comment:
> You seem to like Fill but it does nothing for me, I don't get it.

Fill: "put someone or something into (a space or container) so that it is completely or almost completely full." I wanted a verb, fill came to my mind first and I haven't found a better one yet.